misk-metrics/src/main/kotlin/misk/metrics/MetricsModule.kt

```diff
@@ -23,14 +32,34 @@ class MetricsModule : KAbstractModule() {
     }
   }
 
-  internal class MetricsProvider @Inject constructor(private val v2Metrics: misk.metrics.v2.Metrics) : Provider<Metrics> {
+  internal class MetricsProvider @Inject constructor(private val v2Metrics: V2Metrics) :
+    Provider<Metrics> {
     override fun get(): Metrics {
       return Metrics.factory(v2Metrics)
     }
   }
-  internal class V2MetricsProvider @Inject constructor(private val registry: CollectorRegistry) : Provider<misk.metrics.v2.Metrics> {
-    override fun get(): misk.metrics.v2.Metrics {
-      return misk.metrics.v2.Metrics.factory(registry)
+
+  internal class V2MetricsProvider @Inject constructor(
+    private val registry: CollectorRegistry,
+    private val metricsV3: V3Metrics
+  ) : Provider<V2Metrics> {
+    override fun get(): V2Metrics {
+      return V2Metrics.factory(registry, metricsV3)
+    }
+  }
+
+  internal class MeterRegistryProvider @Inject constructor(
+    private val registry: CollectorRegistry,
+  ) : Provider<PrometheusMeterRegistry> {
+    override fun get(): PrometheusMeterRegistry {
+      return PrometheusMeterRegistry(PrometheusConfig.DEFAULT,registry, Clock.SYSTEM)
```

Comment:
Could we inject the Clock here? Or does using FakeClock in tests cause problems with MeterRegistry?

Reply:
Unfortunately it's a micrometer clock :(